### PR TITLE
DROID-1617 Set | Enhancement | Edit view widget

### PR DESCRIPTION
**Pull request by konstantiniiv**

### CODE REVIEW

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+import com.anytypeio.anytype.core_models.DVViewerType
+import com.anytypeio.anytype.core_models.Id
+import com.anytypeio.anytype.core_models.ObjectTypeIds
+import com.anytypeio.anytype.core_models.ObjectWrapper
+import com.anytypeio.anytype.domain.objects.StoreOfObjectTypes
+import com.anytypeio.anytype.domain.objects.StoreOfRelations
+
+data class ViewerEditWidgetUi(
+    val showWidget: Boolean,
+    val id: Id? = null,
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt:13]: what is id here, if it is nullable?

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/viewer/ViewerView.kt [hunk 2]
```diff
@@ -10,7 +10,11 @@ data class ViewerView(
     val isActive: Boolean,
     val showActionMenu: Boolean = false,
     val isUnsupported: Boolean = false,
-    val defaultObjectType: Id?
+    val defaultObjectType: Id?,
+    val relations: List<String>,
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/viewer/ViewerView.kt:14]: please use typealias here - Key or Id?

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/viewer/ViewerView.kt [hunk 3]
```diff
@@ -10,7 +10,11 @@ data class ViewerView(
...
+    val sorts: List<String>,
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/viewer/ViewerView.kt:15]: please use id typealias here

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/viewer/ViewerView.kt [hunk 4]
```diff
@@ -10,7 +10,11 @@ data class ViewerView(
...
+    val filters: List<String>,
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/viewer/ViewerView.kt:16]: please use typealias here

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    val name: String = "",
+    val viewerId: Id,
+    val defaultObjectType: ObjectWrapper.Type?,
+    val isDefaultObjectTypeEnabled: Boolean = false,
+    val layout: DVViewerType?,
+    val relations: List<String> = emptyList(),
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt:19]: please use typealias - not clear what is string here

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    val filters: List<String> = emptyList(),
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt:20]: idem

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+    val sorts: List<String> = emptyList(),
```
Comment (uburoiubu) [inline presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt:21]: idem

**Review by uburoiubu — APPROVED** [review]

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt (hunk 1 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt:13]: ViewerEditWidgetUi is a state for all settings in any View. If id is empty, then the menu is not valid and is most likely hidden

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/viewer/ViewerView.kt (hunk 3 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/sets/viewer/ViewerView.kt:15]: done

File: presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt (hunk 5 above)
Comment (konstantiniiv) [reply presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ViewerEditWidgetUi.kt:19]: done

**Pull request merged**